fsm/eventbuilder.go

```diff
@@ -0,0 +1,76 @@
+package fsm
+
+type transitionToBuilder struct {
+	name             EventName
+	applyTransition  ApplyTransitionFunc
+	transitionsSoFar map[StateKey]StateKey
+	nextFrom         []StateKey
+}
+
+// To means the transition ends in the given state
+func (t transitionToBuilder) To(to StateKey) EventBuilder {
+	transitions := t.transitionsSoFar
+	for _, from := range t.nextFrom {
+		transitions[from] = to
```

Comment:
In general, what do you think about emitting a warning or error even if we are clobbering previously-set callbacks or transitions?

Reply:
good point!

Reply:
I hate to say it thought I'd rather not mess up the fluidity of the interface I might store the error and fail when you try to construct the event machine which is delayed but preserves the DSL.